src/server/game/Spells/Spell.cpp

```diff
@@ -6392,6 +6398,303 @@ SpellCastResult Spell::CheckPetCast(Unit* target)
+            return delayForDisorients;
+        // Polymorph
+        if (_spell->Id == 12826 || _spell->Id == 12825 || _spell->Id == 12824 || _spell->Id == 118)
+            return delayForDisorients;
+        // Talent - Improved Counter spell silence
+        if (_spell->Id == 18469)
+            return delayForSilences;
+        // Talent - Frostbite
+        if (_spell->Id == 12494)
+            return delayForRoots;
+        // Frostnova
+        if (_spell->Id == 122 || _spell->Id == 865 || _spell->Id == 6131 || _spell->Id == 10230 || _spell->Id == 27088 || _spell->Id == 42917)
+            return delayForRoots;
+        // Pet - Freeze
+        if (_spell->Id == 33395)
+            return delayForRoots;
+            break;
+
+    case SPELLFAMILY_WARLOCK:
+        // Pet - Succubus - Seduction
+        if (_spell->Id == 6358)
+            return delayForBanishes;
+        // Banish
+        if (_spell->Id == 710 || _spell->Id == 18647)
+            return delayForBanishes;
+        // DeathCoil
+        if (_spell->Id == 6789 || _spell->Id == 17925 || _spell->Id == 17926 || _spell->Id == 27223 || _spell->Id == 47859 || _spell->Id == 47860)
+            return delayForHorrors;
+        // Pet - Felguard's intercept
+        if (_spell->Id == 47996)
+            return delayForStuns;
+        // Fear
+        if (_spell->Id == 5782 || _spell->Id == 6213 || _spell->Id == 6215)
+            return delayForFears;
+        // Howl of Terror
+        if (_spell->Id == 5484 || _spell->Id == 17928)
+            return delayForFears;
+        // Shadowfury
+        if (_spell->Id == 30283 || _spell->Id == 30413 || _spell->Id == 30414 || _spell->Id == 47846 || _spell->Id == 47847)
+            return delayForStuns;
+        // Pet - Spell Lock - Silenced
+        if (_spell->Id == 24259)
+            return delayForSilences;
+        break;
+
+    case SPELLFAMILY_DRUID:
+        // Bash
+        if (_spell->Id == 5211 || _spell->Id == 6798 || _spell->Id == 8983)
+            return delayForStuns;
+        // Cyclone
+        if (_spell->Id == 33786)
+            return delayForBanishes;
+        // Hibernate 
+        if (_spell->Id == 2637 || _spell->Id == 18657 || _spell->Id == 18658)
+            return delayForDisorients;
+        // Feral charge - Bear
+        if (_spell->Id == 45334)
+            return delayForInstantSpells;
+        // Feral charge - Cat
+        if (_spell->Id == 49376)
+            return delayForInstantSpells;
+        // Maim
+        if (_spell->Id == 22570 || 49802)
+            return delayForStuns;
+        // Pounce
+        if (_spell->Id == 9005 || _spell->Id == 9823 || _spell->Id == 9827 || _spell->Id == 27006 || _spell->Id == 49803)
+            return delayForOpenerStuns;
+        // Entangling roots
+        if (_spell->Id == 339 || _spell->Id == 1062 || _spell->Id == 5195 || _spell->Id == 5196 || _spell->Id == 9852 || _spell->Id == 9853 || _spell->Id == 26989 || _spell->Id == 53308)
```

Comment:
Entire code could be improved there's no need to check for all of these spells where you could do:

``if (_spell->Mechanic == MECHANIC_ROOT)``

And for stuns

``if (_spell->Mechanic == MECHANIC_STUN)`` And so on and so fourth. Take a look at: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Miscellaneous/SharedDefines.h#L1203

Reply:
That's true but some spells like `Dragon's Breath`, `Frost Nova`, `Mage - Pet - Freeze`, `Felguard's Intercept` and more doesn't have any mechanic assigned, so I prefered to keep the basic functioning by doing case by case.
Plus, there is some spells like hunter traps, or priest's Mind Control that don't require a delay.
I know if there is a delay on `Mind Control` and the targeted player use Repentance at the same time, priest will get 1min cc instead of 10sec.

Reply:
For spells such as Dragon's Breath you can use 
```if (_spell->SpellIconID == 1548)```
Frost nova: 
```MECHANIC_ROOT```
Water Elemental Freeze: 
```if (_spell->Id == 33395)```

It's fine to use Ids for spells that have no mechanics, but using this:
```
// Entangling roots
if (_spell->Id == 339 || _spell->Id == 1062 || _spell->Id == 5195 || _spell->Id == 5196 || _spell->Id == 9852 || _spell->Id == 9853 || _spell->Id == 26989 || _spell->Id == 53308)
```

Instead of this:
```
// Entangling roots 
if (_spell->Mechanic == MECHANIC_ROOT)
```

Is madness.

Reply:
@lineagedr Should be better now